Client/Frontend/Browser/HomePanel/NTPDownloader.swift

```diff
@@ -0,0 +1,215 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+struct NTPItemInfo: Codable {
+    let logoImageUrl: String
+    let logoAltText: String
+    let logoDestinationUrl: URL
+    let logoCompanyName: String
+    let wallpaperImageUrls: [String]
+}
+
+class NTPDownloader {
+    private static let BASE_URL = "https://brave-ntp-crx-input-dev.s3-us-west-2.amazonaws.com/"
+    
+    private let isZipped: Bool
+    private let defaultLocale = "US"
+    private let supportedLocales: [String] = []
+    private let currentLocale = Locale.current.regionCode
+    
+    init(isZipped: Bool = false) {
+        self.isZipped = isZipped
+    }
+    
+    func update() {
+        // Download the NTP Info to a temporary directory
+        self.download { url, error in
+            if let error = error {
+                print(error)
+                return
+            }
+            
+            guard let url = url else {
+                print(error)
+                return
+            }
+            
+            //Move contents of `url` directory
+            //to somewhere more permanent where we'll load the images from..
+            guard let supportDirectory = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first else {
+                return
+            }
+ 
+            do {
+                let downloadsURL = supportDirectory.appendingPathComponent("NTPDownloads")
+                if FileManager.default.fileExists(atPath: downloadsURL.absoluteString) {
+                    try FileManager.default.removeItem(at: downloadsURL)
+                }
+                
+                try FileManager.default.moveItem(at: url, to: downloadsURL)
+            } catch {
+                print(error)
+            }
+        }
+    }
+    
+    func download(_ completion: @escaping (URL?, Error?) -> Void) {
+        if self.isZipped {
+            self.download(path: nil) { [weak self] data, error in
+                guard let self = self else { return }
+                
+                if let error = error {
+                    return completion(nil, error)
+                }
+                
+                guard let data = data else {
+                    return completion(nil, "Invalid Photos.json for NTP Download")
+                }
+                
+                self.unzip(data: data) { url, error in
+                    completion(url, error)
+                }
+            }
+        } else {
+            self.download(path: "photo.json") {[weak self] data, error in
```

Comment:
I think we should add some logic to not re-download if the file hasn't changed on server.
For adblock resources we used etags to determine it

Reply:
Fixed.